### ISSUE TITLE
ruff check with file read from stdin does not use options from pyproject.toml when `--stdin-filename` argument is used

### ISSUE THREAD

**Issue by ipvych** [astral-sh/ruff-lsp#524]
Output of `ruff version`: `ruff 0.11.5 (7186d5e9a 2025-04-10)`

# Steps to reproduce:
```shell
mkdir repro
cd repro
echo '[project]
requires-python = ">=3.11"
[tool.ruff.lint]
extend-select = ["UP"]' > pyproject.toml

mkdir src
cd src
echo -e 'from typing import Callable\nCallable' > repro.py

ruff check --stdin-filename stdin - < repro.py
```

# Observed behavior:
Ruff reported no issues found in file

# Expected behavior:
Pyupgrade plugin that was enabled in pyproject.toml should have reported `Import from collections.abc instead: Callable` error.  Error is correctly reported when I remove `--stdin-filename <anything>` flag from ruff command line or when I run `ruff check repro.py` directly.

**Comment by ipvych** [comment]
Whoops, wrong repo, will open this in ruff repo
